library/std/src/sys/sgx/mod.rs

```diff
@@ -17,14 +17,18 @@ pub mod condvar;
 pub mod env;
 pub mod ext;
 pub mod fd;
+#[path = "../unsupported/fs.rs"]
 pub mod fs;
+#[path = "../unsupported/io.rs"]
 pub mod io;
```

Comment:
It seems a little strange to pull the IoSlice types from somewhere with “unsupported” in the name, as these types are actually used and supported on this target.

Reply:
Yea, good point. I can't think of a good place to put this module, since there isn't a location for "basic implementation that is not Unix".  Would it make sense to add a `sys::basic` module that contained code like this? The only other "functional" code in the `unsupported` tree is:

* [cmath.rs](https://github.com/rust-lang/rust/blob/master/library/std/src/sys/unsupported/cmath.rs) (basic definitions from compiler-builtins)
* common.rs [`strlen`](https://github.com/rust-lang/rust/blob/814d252417cec7cfca98d75db0b3d89555d6d676/library/std/src/sys/unsupported/common.rs#L41-L48) (very simple pure-rust C-strlen implementation)